projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts

```diff
@@ -169,15 +178,53 @@ export class ProductListComponentService {
       .unsubscribe();
   }
 
+  /**
+   * Sort the search results by the given sort code.
+   */
   sort(sortCode: string): void {
-    this.setQueryParams({ sortCode });
+    this.route({ sortCode });
   }
 
-  private setQueryParams(queryParams: SearchCriteria): void {
+  /**
+   * Routes to the next product listing page, using the given `queryParams`. The
+   * `queryParams` support sorting, pagination and querying.
+   *
+   * The `queryParams` are delegated to the Angular router `NavigationExtras`.
+   */
+  protected route(queryParams: SearchCriteria): void {
     this.router.navigate([], {
       queryParams,
       queryParamsHandling: 'merge',
       relativeTo: this.activatedRoute,
     });
   }
+
+  /**
+   * The site context is exposed to update the search query in case of a
```

Comment:
I'm not sure if I understand this correctly - you are saying that the site context is exposed, while the method is private? Maybe I misunderstood.

Reply:
Yes, I moved it back to a private method, as I'm sure this will get refactored in the next iteration. Currently the 2 contexts are hardcoded, while context in general is extensible, by i introducing new context using DI. I've renamed _exposed_ to _used_.